File: src/main/java/com/fish/entity/vo/TopicVO.java
package com.fish.entity.vo;

import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
import com.fasterxml.jackson.databind.annotation.JsonSerialize;
import com.fasterxml.jackson.datatype.jsr310.deser.LocalDateTimeDeserializer;
import com.fasterxml.jackson.datatype.jsr310.ser.LocalDateTimeSerializer;
import com.fish.entity.pojo.Label;
import lombok.AllArgsConstructor;
import lombok.NoArgsConstructor;

import java.io.Serializable;
import java.time.LocalDateTime;
import java.util.ArrayList;

@AllArgsConstructor
@NoArgsConstructor
public class TopicVO implements Serializable {
    private Long topicId;
    private String uploadUserId;
    private String title;
    private String content;
    private String difficulty;
    private Long limitedMemory;
    private Long limitedTime;
    private String inputDescribe;
    private String outputDescribe;
    @JsonSerialize(using = LocalDateTimeSerializer.class)
    @JsonDeserialize(using = LocalDateTimeDeserializer.class)
    private LocalDateTime updateTime;
    private String precautions;
    private String from;
    private ArrayList<Label> labels;

    public Long getTopicId() {
        return topicId;
    }

    public void setTopicId(Long topicId) {
        this.topicId = topicId;
    }

    public String getUploadUserId() {
        return uploadUserId;
    }

    public void setUploadUserId(String uploadUserId) {
        this.uploadUserId = uploadUserId;
    }

    public String getTitle() {
        return title;
    }

    public void setTitle(String title) {
        this.title = title;
    }

    public String getContent() {
        return content;
    }

    public void setContent(String content) {
        this.content = content;
    }

    public String getDifficulty() {
        return difficulty;
    }

    public void setDifficulty(String difficulty) {
        this.difficulty = difficulty;
    }

    public Long getLimitedMemory() {
        return limitedMemory;
    }

    public void setLimitedMemory(Long limitedMemory) {
        this.limitedMemory = limitedMemory;
    }

    public Long getLimitedTime() {
        return limitedTime;
    }

    public void setLimitedTime(Long limitedTime) {
        this.limitedTime = limitedTime;
    }

    public String getInputDescribe() {
        return inputDescribe;
    }

    public void setInputDescribe(String inputDescribe) {
        this.inputDescribe = inputDescribe;
    }

    public String getOutputDescribe() {
        return outputDescribe;
    }

    public void setOutputDescribe(String outputDescribe) {
        this.outputDescribe = outputDescribe;
    }

    public LocalDateTime getUpdateTime() {
        return updateTime;
    }

    public void setUpdateTime(LocalDateTime updateTime) {
        this.updateTime = updateTime;
    }

    public String getPrecautions() {
        return precautions;
    }

    public void setPrecautions(String precautions) {
        this.precautions = precautions;
    }

    public String getFrom() {
        return from;
    }

    public void setFrom(String from) {
        this.from = from;
    }

    public ArrayList<Label> getLabels() {
        return labels;
    }

    public void setLabels(ArrayList<Label> labels) {
        this.labels = labels;
    }
}

File: src/main/kotlin/com/fish/entity/vo/TopicVO.kt
package com.fish.entity.vo

import com.fasterxml.jackson.databind.annotation.JsonDeserialize
import com.fasterxml.jackson.databind.annotation.JsonSerialize
import com.fasterxml.jackson.datatype.jsr310.deser.LocalDateTimeDeserializer
import com.fasterxml.jackson.datatype.jsr310.ser.LocalDateTimeSerializer
import com.fish.entity.pojo.Label
import lombok.AllArgsConstructor
import lombok.NoArgsConstructor
import java.io.Serializable
import java.time.LocalDateTime

class TopicVO(
    var topicId: Long? = null,
    var uploadUserId: String? = null,
    var title: String? = null,
    var content: String? = null,
    var difficulty: String? = null,
    var limitedMemory: Long? = null,
    var limitedTime: Long? = null,
    var inputDescribe: String? = null,
    var outputDescribe: String? = null,
    @JsonSerialize(using = LocalDateTimeSerializer::class)
    @JsonDeserialize(using = LocalDateTimeDeserializer::class)
    var updateTime: LocalDateTime? = null,
    var precautions: String? = null,
    var from: String? = null,
    var labels: ArrayList<Label>? = null,
) : Serializable
